### PR TITLE
EOS-23273/improvements shared storage

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
# Problem Statement
- Added sire nmae to get_path menthod od Storage class.

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/527270615/Shared+Storage

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page : Yet to be done after PR merge

### CODE REVIEW

File: py-utils/src/utils/shared_storage/shared_storage.py [hunk 1]
```diff
@@ -28,19 +29,26 @@ class Storage:
...
+        Conf.load('cotrx_config', Storage.config_file, skip_reload=True)
+        shared_storage_url = Conf.get('cotrx_config', 'support')
 
         self.shared_storage_agent = SharedStorageFactory.get_instance( \
-            shared_storage_type, shared_storage_path)
+            shared_storage_url)
 
     @staticmethod
-    def get_path():
+    def get_path(name: str = None, exist_ok: Bool = True) -> str:
```
Comment (rohit-k-dwivedi) [inline py-utils/src/utils/shared_storage/shared_storage.py:39]: bool should be in all lowercase

File: py-utils/src/utils/shared_storage/shared_storage.py [hunk 2]
```diff
@@ -28,19 +29,26 @@ class Storage:
...
-    def get_path():
+    def get_path(name: str = None, exist_ok: bool = True) -> str:
         """ return shared storage mountpoint """
         storage = Storage()
         shared_path = storage.shared_storage_agent.get_path()
+        if shared_path is None:
+            return None
+        if name:
+            try:
+                spec_path = shared_path + '/' + name + '/'
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/shared_storage/shared_storage.py:47]: use os.path.join()

File: py-utils/src/utils/shared_storage/shared_storage.py [hunk 3]
```diff
@@ -28,19 +29,26 @@ class Storage:
...
+                os.makedirs(spec_path, exist_ok = exist_ok)
+                shared_path = spec_path
+            except OSError as e:
+                raise SharedStorageError(errno.EINVAL, \
+                "dir already exists, (use exist_ok as True) %s" % e)
```
Comment (selvakumaar5496) [inline py-utils/src/utils/shared_storage/shared_storage.py:52]: 4 space indendation

Comment (rahul-tripathi-git) [reply py-utils/src/utils/shared_storage/shared_storage.py:52]: Done

File: py-utils/src/utils/shared_storage/shared_storage.py (hunk 2 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/shared_storage/shared_storage.py]: Done

File: py-utils/src/utils/shared_storage/shared_storage.py (hunk 1 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/shared_storage/shared_storage.py]: Done.

**Review by rohit-k-dwivedi — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by selvakumaar5496 — APPROVED** [review]

**Pull request merged**